src/app/api/dataset.api.ts

```diff
@@ -439,6 +457,40 @@ export class DatasetApi {
             );
     }
 
+    public setVisibilityDataset(params: {
+        accountId: string;
+        datasetId: string;
+        visibility: DatasetVisibilityInput;
+    }): Observable<SetVisibilityDatasetMutation> {
+        return this.setVisibilityDatasetGQL
+            .mutate(
+                {
+                    datasetId: params.datasetId,
+                    visibility: params.visibility,
+                },
+                {
+                    update: (cache) => {
+                        updateCacheHelper(cache, {
+                            accountId: params.accountId,
+                            datasetId: params.datasetId,
+                            fieldNames: ["visibility"],
```

Comment:
Check a case when lineage information depends on visibility - I guess it will need an update too.

Reply:
No, we don't need to do that. I checked that the information is being updated